File: ruby/sum_square_difference.rb
#!/usr/bin/env ruby


# The sum of the squares of the first 100 natural numbers is
n2 = []
1.upto(100) do |x|
  n2 << x**2
end

# The square of the sum of the first 100 natural numbers is
s = []
1.upto(100) do |x|
  s << x
end

total = 0
s.each do |x|
  total += x
end

puts total**2 - n2.inject(:+)

File: python/sum_square_difference.py
#!/usr/bin/env python
"""
Find the difference between the sum of the squares of the first one hundred natural numbers and the square of the sum.
"""


def main():

    # The sum of the squares of the first 100 natural numbers is
    n2 = sum([ (x+1)**2 for x in range(0, 100)])

    # The square of the sum of the first 100 natural numbers is
    s = sum([ x + 1 for x in range(0, 100)])**2

    print s - n2

if __name__ == "__main__":
    main()